addons/mass_mailing/models/mass_mailing.py

```diff
@@ -315,6 +316,20 @@ def _get_source_id(self, cr, uid, context=None):
     def _get_medium_id(self, cr, uid, context=None):
         return self.pool['ir.model.data'].xmlid_to_res_id(cr, uid, 'utm.utm_medium_email')
 
+    # To be removed in saas-8
+    def create(self, cr, uid, vals, context=None):
+        if self.pool['utm.campaign'].browse(cr, uid, vals['campaign_id'], context=context).name != vals['name']:
```

Comment:
If campaign_id is given we should keep it whatever the name. Moreover you use vals['campaign_id'] that may not exist (crash).

I would instead use
´´´if not vals.get('campaign_id'):
    vals['campaign_id'] = self.pool[].create(...)´´´